src/components/calculator.js

```diff
@@ -1,51 +1,118 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+      operation: null,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
   }
 
+  handleChange(e) {
+    this.setState({ next: e.target.value });
+  }
+
+  handleClick(e) {
+    if (this.isNumber(e.target.outerText)) {
+      console.log('num');
+      this.setState({ next: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+      console.log(res);
+    }
+
+    if (e.target.outerText === '+' || e.target.outerText === '-' || e.target.outerText === 'x' || e.target.outerText === '÷') {
+      this.setState({ operation: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+      this.setState({ total: res.total });
+      console.log(res);
+      console.log(this.state);
+    }
+
+    if (e.target.outerText === '=') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.total });
+      console.log(res);
```

Comment:
* I suggest you remove `console.log` from your codebase as it is not professional and might log sensitive information to  console therefore exposing it to the user [Read more](https://medium.com/@hfally/deactivate-console-log-on-production-why-and-how-d0345abbe71)